troughs_and_waves/gaussian/tests/test_gaussian.py

```diff
@@ -0,0 +1,51 @@
+#  (C) Copyright 2010-2020 Enthought, Inc., Austin, TX
+#  All rights reserved.
+
+import unittest
+
+from unittest import mock
+
+from traits.testing.api import UnittestTools
+from force_bdss.api import DataValue, Slot, BaseDataSourceFactory
+
+from troughs_and_waves.gaussian.gaussian import Gaussian
+from troughs_and_waves.gaussian.gaussian_model import GaussianModel
+
+
+class TestGaussian(unittest.TestCase, UnittestTools):
+    def setUp(self):
+        self.factory = mock.Mock(spec=BaseDataSourceFactory)
+
+    def test_initialization(self):
+        ds = Gaussian(self.factory)
+        self.assertEqual(ds.factory, self.factory)
+
+    def test_run(self):
+        ds = Gaussian(self.factory)
+        model = GaussianModel(self.factory)
```

Comment:
We can use

```python
ds = self.factory.create_data_source()
model = self.factory.create_model()
```

here in order to test the integration of factory, data source and model classes